### ISSUE TITLE
Make 'Add' buttons consistent across data views, simplify labels

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#58198]
<img width="1612" alt="add" src="https://github.com/WordPress/gutenberg/assets/846565/b954a998-88b6-487c-b1f1-9927bb7b3388">

- [ ] "Add **new** $post_type" is unnecessarily verbose. We can shorten this to "Add $post_type". Trac ticket: https://core.trac.wordpress.org/ticket/61219#ticket
- [x] The add flow for Patterns should match other data views. IE a primary button in the page header. For now this button will open the same menu as the current `+` in the sidebar. The `+` button can be removed.

**Comment by afercia** [comment]
> The labels are unnecessarily long. "Add new" would suffice in all cases rather than the verbose "Add New Template Part".

You may not be aware that recently in core all the 'Add new' strings have been expanded to include the _what_ they refer to. This has been done for accessibility reasons _and_ to be consistent with Gutenberg.

I'm not sure this should be changed. Please don't shorten to 'Add new'.

Cc @joedolson @alexsanford

**Comment by jameskoster** [comment]
@afercia couldn't we use aria labels like the action links in wp-admin tables?

<img width="744" alt="Screenshot 2024-01-24 at 16 06 19" src="https://github.com/WordPress/gutenberg/assets/846565/00698192-20f2-4bd3-a318-d9a35b1a4364">

**Comment by joedolson** [comment]
I'd really like to avoid using different logic in new interfaces from what we've decided to use in core. In core, we've agreed that these controls need to include context, so let's not just casually reverse that in a new interface. 

Regarding the row actions, those use aria-label only because of the amount of repetition and the possible length. There are several row actions, and each one needs that context included, and the post titles could be quite a bit longer than a single word.

"Add New Template Part" is a nominal amount of text.

In row actions, we'd potentially be looking at `Edit "You only live once: the history of frisbees in Wollongong" | Quick Edit "You only live once: the history of frisbees in Wollongong"`, etc. You get the point. Visually, that starts to take up an enormous amount of space and repetition.

Screen reader users don't necessarily have to experience that entire text; spoken text can be interrupted, unlike visual text. So a screen reader navigating through these same links only needs to hear enough to verify that the context is what they expected it to be. So that quantity of text is less of an interference for screen reader users than it is for sighted users.

**Comment by jameskoster** [comment]
At the least, now the the Patterns data view is stable, we should update the "Add new" UX there to match other data views. What do you think @oandregal @ntsekouras ?

**Comment by ntsekouras** [comment]
>At the least, now the the Patterns data view is stable, we should update the "Add new" UX there to match other data views.

You mean a new `add pattern + add template part` button at the right top corner of the UI? Since in this page we can create two different things, should be a simple `Add new`(or Create new) button that opens the same dialog with the three options(including the import pattern)?

**Comment by jameskoster** [comment]
> You mean a new add pattern + add template part button at the right top corner of the UI? 

That's correct. A button that opens the same dialog is fine. 

Since template parts and patterns are converging we should probably consolidate the template part / pattern creation modals into a single UX. Importing would be a secondary action. So in the future we'd end up with something like:

<img width="391" alt="Screenshot 2024-01-30 at 12 29 34" src="https://github.com/WordPress/gutenberg/assets/846565/e2905780-16af-4491-8392-875a8cee1da4">

Until then perhaps the label can read "Create new..." with the ellipsis indicating subsequent action is required. What do you think?

**Comment by pablohoneyhoney** [comment]
I agree with @afercia on this https://github.com/WordPress/gutenberg/issues/58198#issuecomment-1908302581

UX wise, it's better to advance the value created within the button itself. 
I'm not sure how that applies across all logics and cases, since I also agree `Add new template part` is quite a long CTA. But maybe it's a better solution to remove `new`, since `Add` provides that additive context already.

**Comment by jameskoster** [comment]
I've updated the OP based on the discussion here to make the issue actionable.

**Comment by youknowriad** [comment]
I think the first part of this issue should be a trac ticket as it means changing the "add_new_item" label for all the core post types and the post type registration is maintained on Core.

**Comment by jameskoster** [comment]
Re-opening this to track the outstanding issue.

**Comment by im3dabasia** [comment]
The first part has been addressed and merged into trunk in upstream repo in Feb '25, and the [ticket](https://core.trac.wordpress.org/ticket/61219#ticket) has been marked as completed. 

Since the overall work related to this issue has been completed I think we can close this issue? 

Looping in @jameskoster
